File: Bridge/cpp/CivilianTimeImp.cpp
#include "CivilianTimeImp.h"

CivilianTimeImp::CivilianTimeImp(int hr, int min, int pm): TimeImp(hr, min) {
	if (pm){
	  strcpy(whichM_, " PM");
	}else{
	  strcpy(whichM_, " AM");
	 }
}

/* virtual */
void CivilianTimeImp::tell() {
	cout << "time is " << hr_ << ":" << min_ << whichM_ << endl;
}

File: Bridge/cpp/java/src/dp/bridge/time/CivilianTimeImp.java
package dp.bridge.time;

public class CivilianTimeImp extends TimeImp {
	protected String whichM;
	
	public CivilianTimeImp(int hr, int min, int pm){
		super(hr, min);
		if(pm == 1){
			this.whichM = " PM";
		} else {
			this.whichM = " AM";
		}
	}
	
	public void tell(){
		System.out.println("time is " + this.hr + ":" + this.min + this.whichM);
	}
}